subgraphs/aave-forks/src/mapping.ts

```diff
@@ -372,707 +511,590 @@ export function _handleReserveDataUpdated(
+
+  const premiumUSDTotal = tokenManager.getAmountUSD(premiumAmount);
+  let premiumUSDToProtocol = premiumUSDTotal.times(protocolRevenueShare);
+  let premiumUSDToLP = premiumUSDTotal.minus(premiumUSDToProtocol);
+  let premiumUSDToDeduct = premiumUSDTotal;
+  const premiumRateTotal = flashloanPremium.premiumRateTotal;
+  let premiumRateToProtocol = premiumRateTotal.times(protocolRevenueShare);
+  let premiumRateToLP = premiumRateTotal.minus(premiumRateToProtocol);
+
+  if (flashloanPremium.premiumRateToProtocol.gt(BIGDECIMAL_ZERO)) {
+    // according to https://github.com/aave/aave-v3-core/blob/29ff9b9f89af7cd8255231bc5faf26c3ce0fb7ce/contracts/interfaces/IPool.sol#L634
+    // premiumRateToProtocol is the percentage of premium to protocol
+    premiumUSDToProtocol = premiumUSDTotal
+      .times(flashloanPremium.premiumRateToProtocol)
+      .plus(premiumUSDToProtocol);
+    premiumRateToProtocol = premiumRateTotal
+      .times(flashloanPremium.premiumRateToProtocol)
+      .plus(premiumRateToProtocol);
+
+    // premium to LP
+    premiumUSDToLP = premiumUSDTotal.minus(premiumUSDToProtocol);
+    premiumRateToLP = premiumRateTotal.minus(premiumRateToProtocol);
+    // this part of the premium is transferred to the treasury and not
+    // accrued to liquidityIndex and thus no need to deduct
+    premiumUSDToDeduct = premiumUSDToDeduct.minus(
+      premiumUSDTotal.times(flashloanPremium.premiumRateToProtocol)
+    );
+  }
 
-  // update usage metrics
-  snapshotUsage(
-    protocol,
-    event.block.number,
-    event.block.timestamp,
-    liquidate.liquidatee,
-    EventType.LIQUIDATEE,
-    true // only count this liquidate as new tx
+  const feeToProtocol = manager.getOrUpdateFee(
+    FeeType.FLASHLOAN_PROTOCOL_FEE,
+    null,
+    premiumRateToProtocol
   );
-  snapshotUsage(
-    protocol,
-    event.block.number,
-    event.block.timestamp,
-    liquidate.liquidator,
-    EventType.LIQUIDATOR, // updates dailyActiveLiquidators
-    false
+  manager.addProtocolRevenue(premiumUSDToProtocol, feeToProtocol);
+
+  // flashloan premium to LP is accrued in liquidityIndex and handled in
+  // _handleReserveDataUpdated;
+  // https://github.com/aave/aave-v3-core/blob/master/contracts/protocol/libraries/logic/FlashLoanLogic.sol#L233-L237
+  const feeToLP = manager.getOrUpdateFee(
+    FeeType.FLASHLOAN_LP_FEE,
+    null,
+    premiumRateToLP
   );
 
-  // udpate market daily / hourly snapshots / financialSnapshots
-  updateSnapshots(
-    protocol,
-    market,
-    liquidate.amountUSD,
-    EventType.LIQUIDATOR,
-    event.block.timestamp,
-    event.block.number
-  );
+  manager.addSupplyRevenue(premiumUSDToLP, feeToLP);
+  // save premiumUSDToDeduct to be deducted from total revenue to avoid double counting
+  flashloanPremium.premiumUSDToDeduct = premiumUSDToDeduct;
+  flashloanPremium.save();
 }
 
 /////////////////////////
 //// Transfer Events ////
 /////////////////////////
-
 export function _handleTransfer(
```

Comment:
won't we need to add a flag to this for the debt type since we can have transfers of stokens and vtokens and they affect different pos entities

Reply:
I think you're right with my accountDebtBalance implementation. If we move to use Transfer event, then this would be as it is, correct?

Reply:
Yeah I think I never added this into the SDK when I originally made it, but there should be an optional field in `createTransfer` for the type... Right?

Reply:
@tnkrxyz actually `manager.createTransfer()` does have an optional field for `interestRateType` why do we not use this here?

Reply:
fixed.